core/shared/src/main/scala-2.11/zio/CanFail.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.annotation.implicitNotFound
+
+/**
+ * A value of type `CanFail[E]` provides implicit evidence that an effect with
+ * error type `E` can fail, that is, that `E` is not equal to `Nothing`.
+ */
+@implicitNotFound("This operation only makes sense for effects that can fail.")
+sealed trait CanFail[-E]
+
+object CanFail extends CanFail[Any] {
+
+  implicit final def canFail[E]: CanFail[E] = CanFail
+
+  implicit final val canFailAmbiguous1: CanFail[Nothing] = CanFail
+  implicit final val canFailAmbiguous2: CanFail[Nothing] = CanFail
```

Comment:
For my own understanding, why is there two such values? Also it may be useful to add an explanatory comment for the future.